docs/proposals/alternate-api-representations.md

```diff
@@ -0,0 +1,422 @@
+knows apply. The server must return a structured resource usable by both command line and
+rich clients (web or IDE), which implies a schema, which implies JSON, and which means the
+server should return a known Kind. For this example we will call that kind `TabularOutput`
+to demonstrate the concept.
+
+A server side resource would implement a transformation from their resource to `TabularOutput`
+and the API machinery would translate a single item or a list of items (or a watch) into
+the tabular resource.
+
+A generic client wishing to display a tabular list for resources of type `v1.ReplicaSets` would
+make the following call:
+
+    GET /api/v1/namespaces/example/replicasets
+    Accept: application/json;g=api.k8s.io,v=v1,as=TabularOutput, application/json
+
+    200 OK
+    Content-Type: application/json;g=api.k8s.io,v=v1,as=TabularOutput
+    {
+      "apiVersion": "api.k8s.io/v1",
+      "kind": "TabularOutput",
+      "columns": [
+        {"name": "Name", "description": "The name of the resource"},
+        {"name": "Resource Version", "description": "The version of the resource"},
+        ...
+      ],
+      "items": [
+        {"columns": ["name", "10", ...]},
+        ...
+      ]
+    }
+
+The client can then present that information as necessary. If the server returns the
+resource list `v1.ReplicaSetList` the client knows that the server does not support tabular
+output and so must fall back to a generic output form (perhaps using the existing
+compiled in listers).
+
+Note that `kubectl get` supports a number of parameters for modifying the response,
+including whether to filter resources, whether to show a "wide" list, or whether to
+turn certain labels into columns. Those options are best represented as query parameters
+and transformed into a known type.
+
+
+### Example: Versioning a ListOptions call to a generic API server
+
+When retrieving lists of resources, the server transforms input query parameters like
+`labels` and `fields` into a `ListOptions` type. It should be possible for a generic
+client dealing with the server to be able to specify the version of ListOptions it
+is sending to detect version skew.
+
+Since this is an input and list is implemented with GET, it is not possible to send
+a body and no Content-Type is possible.  For this approach, we recommend that the kind
+and API version be specifiable via the GET call for further clarification:
+
+New query parameters:
+
+| Name | Value | Default | Description |
+| ---- | ----- | ------- | ----------- |
+| kind | The kind of parameters being sent | `ListOptions` (GET), `DeleteOptions` (DELETE) | The kind of the serialized struct, defaults to ListOptions on GET and DeleteOptions on DELETE. |
+| apiVersion | The API version of the parameter struct | `api.k8s.io/v1` | May be altered to match the expected version.  Because we have not yet versioned ListOptions, this is safe to alter. |
```

Comment:
groupVersion? Would like to avoid making more things called apiVersion when they actually include the group, too.

also I'd argue again for `v1.core.k8s.io`.